### PR TITLE
Backport the DB2 client to Vert.x 3.9

### PR DESCRIPTION
**Pull request by aguibert**
Brings the DB2 client driver back to the Vert.x 3.9 branch. Functionality of the DB2 driver in the 3.9 branch will be on par with the 4.0 counterpart, except for any differences at the common Sql Client API level (e.g. `Future`ised client).

## Motivation

In order to improve stability of the DB2 client, we need to get it into users hands in a released version of projects. Currently work is going on in HibernateRX to build on top of the 3.9 client, and on top of that work is going on to pull HibernateRX into Quarkus. By backporting the DB2 driver to 3.9, users of HibernateRX and Quarkus will be able to use the DB2 client and I can use that feedback to improve quality of the DB2 client.

## Note to reviewers:
The initial copy/paste of the 4.0 code is in the first commit, and all commits after that represent changes needed to adapt the 4.0 code to be compatible with 3.9.

### CODE REVIEW

**Comment by vietj** [comment]
@aguibert did you check file formatting ? I've seen in master many 4 spaces used for indenting instead of 2, this PR looks correct but I cannot check everything. We should also reformat DB2 source in master to ease potential cherry picks between branches.

**Comment by aguibert** [comment]
@vietj I try to keep 2 spaces where I can (because I know that's the conventions this repo uses) but in some cases 4 spaces / tabs slip through. Do you know of any auto-formatting tools we could use? In Quarkus they have source formatting as part of the build process which is quite nice.

Also, there is still 1 failure that I need to look into before we can merge this. The `PgConnectionTestBase.testTransactionAbort()` test case is failing, so I need to look into that more and get it fixed.

**Comment by pmlopes** [comment]
@aguibert maybe you missed https://github.com/vert-x3/wiki/wiki/Vert.x-code-style-guidelines I've added a editorconfig file there. If you use it and your ide supports it it should be easy to keep the spacing config on the project.

**Comment by aguibert** [comment]
@vietj I fixed the failing PG test so now the CI build is green. The main thing to check for is the changes to TransactionImpl here:
https://github.com/eclipse-vertx/vertx-sql-client/pull/611/files#diff-e42176acb733da02278e657123ea3d5f

**Comment by aguibert** [comment]
@vietj this one is ready for re-review now. I've applied source code formatting in this PR and also on the master branch, so things should be easier for you to diff now. I also included the DB2 portion of the Driver PR

File: pom.xml [hunk 1]
```diff
@@ -106,6 +106,7 @@
     <module>vertx-sql-client</module>
     <module>vertx-pg-client</module>
     <module>vertx-mysql-client</module>
+	<module>vertx-db2-client</module>
```
Comment (vietj) [inline pom.xml:109]: incorrect indent.

File: pom.xml [hunk 2]
```diff
@@ -59,7 +59,7 @@
     <!-- We skip sources jar generation as we do it with the assembly plugin to have greater control over the content -->
     <source.skip>true</source.skip>
     <kotlin.version>1.3.0</kotlin.version>
-    <testcontainers.version>1.12.0</testcontainers.version>
+    <testcontainers.version>1.12.4</testcontainers.version>
```
Comment (vietj) [inline pom.xml:62]: we cannot upgrade as this requires a CQ for this version

**Comment by aguibert** [comment]
@vietj ok, I downgraded us back to the original Testcontainers version and fixed the formatting. Should be all set for approval now

**Comment by vietj** [comment]
@aguibert LGTM

**Pull request merged**